### PR TITLE
Expose varying stripe granularity from Dask

### PR DESCRIPTION
**Pull request by calebwin**
This exposes `stripe_size_bytes`, `stripe_size_rows`, and `row_index_stride` (these are parameters that were added to `cudf.to_orc` by https://github.com/rapidsai/cudf/pull/9310/) to `dask_cudf.to_orc`.

### CODE REVIEW

File: python/dask_cudf/dask_cudf/io/orc.py [hunk 1]
```diff
@@ -115,12 +115,28 @@ def read_orc(path, columns=None, filters=None, storage_options=None, **kwargs):
     return dd.core.new_dd_object(dsk, name, meta, divisions)
 
 
-def write_orc_partition(df, path, fs, filename, compression=None):
+def write_orc_partition(
+    df,
+    path,
+    fs,
+    filename,
+    compression,
```
Comment (charlesbluca) [inline python/dask_cudf/dask_cudf/io/orc.py:123]: Are we safe to remove the default value for this argument?

File: python/dask_cudf/dask_cudf/io/orc.py [hunk 2]
```diff
@@ -146,6 +165,15 @@ def to_orc(
     storage_options: None or dict
         Further parameters to pass to the bytes backend.
     compression : string or dict, optional
+    stripe_size_bytes: integer or None, default None
+        Maximum size of each stripe of the output.
+        If None, 67108864 (64MB) will be used.
+    stripe_size_rows: integer or None, default None 1000000
+        Maximum number of rows of each stripe of the output.
+        If None, 1000000 will be used.
+    row_index_stride: integer or None, default None 10000
```
Comment (charlesbluca) [inline python/dask_cudf/dask_cudf/io/orc.py:174]: Nitpicky, but feel like we can remove the implicit default value as long as it is mentioned in the param description:

```suggestion
    stripe_size_rows: integer or None, default None
        Maximum number of rows of each stripe of the output.
        If None, 1000000 will be used.
    row_index_stride: integer or None, default None
```

**Review by charlesbluca — COMMENTED** [review]
Thanks for the work here @calebwin! Were you able to find a test that could be parametrized to use these new params?

File: python/dask_cudf/dask_cudf/io/orc.py (hunk 2 above)
Comment (calebwin) [reply python/dask_cudf/dask_cudf/io/orc.py]: Done, thanks

**Comment by calebwin** [comment]
@charlesbluca Thanks for the review. I've parameterized the `to_orc` test to test this but I had to change the number of rows in the ORC dataset written in each parametrization to be 30,000 (because the cuDF implementation requires stripe size to be at least 64 KiB or 512 rows) - do you think that's okay?

File: python/dask_cudf/dask_cudf/io/orc.py (hunk 1 above)
Comment (calebwin) [reply python/dask_cudf/dask_cudf/io/orc.py:123]: There is no default value here now.

**Comment by vyasr** [comment]
@charlesbluca is this something that we want in for 21.12, or can we move to 22.02? If it's for 21.12 could you request appropriate reviewers (for some reason there aren't any here despite it not being a draft)?

**Comment by charlesbluca** [comment]
Not sure, @calebwin was there any larger issue this was opened to resolve?

**Comment by calebwin** [comment]
@charlesbluca @vyasr This was just so I could benchmark varying stripe strize in ORC in cuDF as well as with Dask cuDF for larger datasets. There was no specific issue that this is addressing but Randy did say that there was interest in being able to vary the granularity of stripes in ORC datasets (hence https://github.com/rapidsai/cudf/pull/9310) and the interest was coming from a Dask cuDF user. So presumably they would want to be able to set the stripe size when using Dask cuDF's `to_orc`.

**Comment by vyasr** [comment]
Got it. I'm sure there's interest, but we're hitting code freeze today so since there doesn't appear to be a truly urgent driver I'm going to recommend pushing this to 22.02 if that's OK with both of you.

**Comment by jakirkham** [comment]
Moved to 22.02

**Comment by vyasr** [comment]
@randerzander I assume based on #9310 that this is something that was desired? Should we try to finish up this PR?

**Comment by vyasr** [comment]
I'm going to close this PR for now, but please feel free to reopen if there's still interest in implementing this feature.

**Pull request closed without merging**